apps/entity-service/src/file/file-upload.service.ts

```diff
@@ -51,11 +60,15 @@ const VALIDATION = {
 export class FileUploadService {
   private logger = new TMLogger(FileUploadService.name);
 
-  constructor(private readonly mediaService: MediaService, protected readonly entitiesService: EntitiesService) {}
-
-  public async uploadFile(model: EntityModel, entity: EntityType, collection: string, req: FastifyRequest) {
-    const { file, ...extraFields } = await this.extractRequestData(req);
+  constructor(private readonly mediaService: MediaService, private readonly entitiesService: EntitiesService) {}
 
+  public async uploadFile(
+    model: EntityModel,
+    entity: EntityType,
+    collection: string,
+    file: Express.Multer.File,
```

Comment:
This, combined with the naked `import "multer";` above is a bit weird. Is it possible to something more like `import { File } from "multer";` and avoid accessing the type from the global express namespace? Same question for the controller.

Reply:
there are some problems with the typing otherwise, I will give another look, but for now this is the way it works without type errors